pkg/remote/aws/repository/iam_repository.go

```diff
@@ -0,0 +1,208 @@
+package repository
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/iam/iamiface"
+)
+
+type IAMRepository interface {
+	ListAllAccessKeys() ([]*iam.AccessKeyMetadata, error)
+	ListAllUsers() ([]*iam.User, error)
+	ListAllPolicies() ([]*iam.Policy, error)
+	ListAllRolePolicies() ([]string, error)
+	ListAllRolePolicyAttachments() ([]*AttachedRolePolicy, error)
+	ListAllRoles() ([]*iam.Role, error)
+	ListAllUserPolicyAttachments() ([]*AttachedUserPolicy, error)
+	ListAllUserPolicies() ([]string, error)
+}
+
+type iamRepository struct {
+	client iamiface.IAMAPI
+}
+
+func NewIAMClient(session *session.Session) *iamRepository {
+	return &iamRepository{
+		iam.New(session),
+	}
+}
+
+func (r *iamRepository) ListAllAccessKeys() ([]*iam.AccessKeyMetadata, error) {
+	users, err := r.ListAllUsers()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
@eliecharra @moadibfr wonder what you think about this. Should we call one repository method inside an another one ? And if so, what about the error message that we used before when the status code is 403 ?

Reply:
I think it makes sense since the method name there is `ListAllAccessKeys`.
There is two approach IMHO, the actual one and another when every repository method are independent.
So we can imagine two methods `ListUsers() []User` and `ListAccessKeys(user User) []AccessKey`.
Then in the supplier code we first call user enumeration, the in a loop we'll call `ListAccessKeys`.

I'm not sure about the pro and cons of the two approach, maybe the second one is easier to test and is better in term of separation of concern ?

Reply:
I had in mind you're second one indeed where the `ListAllAccessKeys` method has a `User` argument. We could easily test the 403 in the first call of `ListAllUsers` and then a 403 in the second call inside the supplier.

Reply:
But the second one was the previous methodology used inside suppliers. I was thinking this PR was about changing this behaviour

Reply:
By calling the `ListAllUsers` inside the `ListAllAccessKeys`, you made the latter depend on the former one. And IMO, you made it harder to test with 403 errors. Just look at your testz, we had 2 differents tests for errors. One originated from the `ListAllUsers`, where we output the error came from the listing of users but inside the listing of the access keys. And the second one where this is in fact the listing of the access keys that results in a 403. With what you did, you were obliged to delete the first test.

Reply:
Modification done.